test/e2e/http2_cert_test.go

```
@@ -0,0 +1,153 @@
+
+	err := wait.Poll(1*time.Second, framework.AsyncOperationTimeout, func() (bool, error) {
+		route, getErr := client.Routes.Routes(api.OpenShiftConsoleNamespace).Get(context.TODO(), api.OpenShiftConsoleRouteName, metav1.GetOptions{})
+		if getErr != nil {
+			return false, getErr
+		}
+		if route.Spec.TLS == nil || len(route.Spec.TLS.Certificate) == 0 {
+			return false, nil
+		}
+		adminSecret, secretErr := client.Core.Secrets(api.OpenShiftConfigNamespace).Get(context.TODO(), tlsSecretName, metav1.GetOptions{})
+		if secretErr != nil {
+			return true, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 125) but it returns nil

(nilerr)